react-app/src/utils/textHelpers.ts

```diff
@@ -12,8 +12,20 @@ export function removeUnderscoresAndCapitalize(
     .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
     .join(" ");
 
-  // Remove 's' from the end if it exists
-  return capitalized.endsWith("s") ? capitalized.slice(0, -1) : capitalized;
+  
+  const words = capitalized.split(" ");
+  
+  // Remove 's' from the first word if it exists
+  if (words.length > 0 && words[0].endsWith("s")) {
+    words[0] = words[0].slice(0, -1);
+  }
+  
+  // Remove 's' from the end of the final word if it exists
+  if (words.length > 0 && words[words.length - 1].endsWith("s")) {
+    words[words.length - 1] = words[words.length - 1].slice(0, -1);
+  }
+  
+  return words.join(" ");
```

Comment:
Could you please explain what's happening here? Trying to reason about the reassignments is confusing

Reply:
When I was testing  `expect(removeUnderscoresAndCapitalize("cats_and_dogs")).toBe("Cat And Dog");` return false because in textHelper.ts, the original lines 15-16 would only remove s from the last word. It would return cats and dog, instead of cat and dog. I wanted to remove the s from both cats and dogs. I can change it back and try editing the test.

Reply:
switched it back